jenkins/L0_Test.groovy

```
@@ -920,16 +1010,33 @@ def runLLMTestlistOnPlatformImpl(pipeline, platform, testList, config=VANILLA_CO
                 string(credentialsId: 'llm_evaltool_repo_url', variable: 'EVALTOOL_REPO_URL')
             ]) {
                 sh "env | sort"
-                trtllm_utils.llmExecStepWithRetry(
-                    pipeline,
-                    numRetries: 1,
-                    script: """
-                        rm -rf ${stageName}/ && \
-                        cd ${llmSrc}/tests/integration/defs && \
-                        ${testCmdLine.join(" ")}
-                    """,
-                    retryLog: "stageName = ${stageName}, HOST_NODE_NAME = ${env.HOST_NODE_NAME}"
-                )
+                if (env.alternativeTRT || testFilter[(IS_POST_MERGE)]) {
+                    trtllm_utils.llmExecStepWithRetry(
+                        pipeline,
+                        numRetries: 1,
+                        script: """
+                            rm -rf ${stageName}/ && \
+                            cd ${llmSrc}/tests/integration/defs && \
+                            ${testCmdLine.join(" ")}
+                        """,
+                        retryLog: "stageName = ${stageName}, HOST_NODE_NAME = ${env.HOST_NODE_NAME}"
+                    )
+                } else {
+                    try {
+                        sh """
+                            rm -rf ${stageName}/ && \
+                            cd ${llmSrc}/tests/integration/defs && \
+                            ${testCmdLine.join(" ")}
+                        """
+                    } catch (InterruptedException e) {
+                        throw e
+                    } catch (Exception e) {
+                        isRerunFailed = rerunFailedTests(stageName, llmSrc, extraInternalEnv, pytestTestTimeout)
+                        if (isRerunFailed) {
+                            error "Rerun tests failed"
```

Maybe here is mean the tests still failed after rerun?

---

Yes

---

Feel like this needs to be a little more readable, and if it's an error in pytest's footsteps, it makes more sense to print a msg like "The tests keep failed after rerun" and throw e (keep raw error message)?

---

We should throw the original error or error in rerun step?

---

Do you have a pipeline that displays this log? I'm not sure if error xxx generates a separate step, but if not, error xxx is reasonable